extension/jaegerremotesampling/internal/grpc.go

```diff
@@ -99,6 +99,7 @@ func (s *SamplingGRPCServer) Shutdown(ctx context.Context) error {
 	select {
 	case <-ctx.Done():
 		s.grpcServer.Stop()
+		<-ch // wait for the graceful stop goroutine to return
```

Comment:
what is the pattern used by OTLP server in main collector repo? We should use the same mechanisms

Reply:
The only grpc implementation I found was in the otlp receiver [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/otlp.go#L186-L199). It uses a waiting group which gets complete when the server is actually stopped after calling` GracefulStop()`. So, it does not take into account the context timeout as found in our code. Should I change the implementation to match this?

Reply:
sgtm - if it works for the exporter which is most likely using in 100% of configurations, why does this one needs to be more elaborate for shutdown.

Reply:
Pushed the changes. Do we need to tweak the tests?